core/templates/pages/exploration-editor-page/modal-templates/lost-changes-modal.component.spec.ts

```diff
@@ -135,7 +116,27 @@ describe('Lost Changes Modal Component', () => {
     fixture.detectChanges();
 
     expect(modalBody).toBe(
-      'The lost changes are displayed below. You may want to copy and ' +
-      'paste these changes before discarding them.');
+      'The lost changes are displayed below. You may want to export or ' +
+      'copy and paste these changes before discarding them.');
+  });
+
+  it('should export the lost changes and close the modal', () => {
+    spyOn(
+      fixture.elementRef.nativeElement, 'getElementsByClassName'
+    ).withArgs('oppia-lost-changes').and.returnValue([
+      {
+        innerText: 'Dummy Inner Text'
+      }
+    ]);
+    const dismissSpy = spyOn(ngbActiveModal, 'dismiss').and.callThrough();
+    const spyObj = jasmine.createSpyObj('a', ['click']);
+    spyOn(document, 'createElement').and.returnValue(spyObj);
+    component.exportChangesAndClose();
+    expect(document.createElement).toHaveBeenCalledTimes(1);
+    expect(document.createElement).toHaveBeenCalledWith('a');
+    expect(spyObj.download).toBe('lostChanges.txt');
+    expect(spyObj.click).toHaveBeenCalledTimes(1);
+    expect(spyObj.click).toHaveBeenCalledWith();
+    expect(dismissSpy).toHaveBeenCalledWith();
```

Comment:
Please make sure this test suite does not result in flakiness (we seem to be using global "document" in this test). Feel free to talk to @aishwary023 if you need any help in verifying this.

Reply:
Will do this.